src/optimize.c

```diff
@@ -1175,12 +1176,17 @@ Expression *Expression_optimize(Expression *e, int result, bool keepLvalue)
             }
             if (e->e1->isBool(true))
             {
-                // Replace with (e1, 1)
-                ret = new CommaExp(e->loc, e->e1, new IntegerExp(e->loc, 1, e->type));
-                ret->type = e->type;
+                // Replace with true
+                ret = new IntegerExp(e->loc, 1, Type::tbool);
+                if (e->type->toBasetype()->ty == Tvoid)
+                {
+                    ret = new CastExp(e->loc, ret, Type::tvoid);
+                    ret->type = e->type;
+                }
```

Comment:
If `e->e1` has runtime side effects, `e->e1->isBool(true)` returns false. In other words, `e->e1` is already optimized to a trivial constant in here.